### ISSUE TITLE
Constant scoping issues

### ISSUE THREAD

**Issue by kliuless** [egonSchiele/contracts.ruby#77]
This looks like a very promising lib, but I get `NameError`'s when a contract is violated.
I'm on ruby 2.1.2.

Using the version from rubygems.org (0.5), I get this error:

    contracts-0.5/lib/contracts.rb:134:in `failure_msg': uninitialized constant Contract::Support (NameError)

Using master (26de665a), I get this error:

    lib/contracts.rb:247:in `call_with': uninitialized constant Contract::Func (NameError)

It looks like the `Contract` class is trying to access constants inside `Contracts` (plural) without specifying the scope via `::`. Is there a stable version I can use?

Also, I think it would be helpful to have tags & branches for each release so it's easier to view changes between releases.

**Comment by waterlink** [comment]
You can try `0.4`. Should work, because `Contract::Support` was introduced after it. You are right about tags though.

Are you using any heavy framework, like `Rails` or something similar?

Can you provide a failing example, along these lines, it will be very helpful:

```ruby
require "contracts"

class MyExample
  # .. do definitions with contracts ..
end

# .. misuse MyExample, so that it fails to make a contract violation
```

If it is not failing in such a simple setup, then I will probably need some more info on your setup.

**Comment by waterlink** [comment]
@kliuless thx for reporting, looks like I have pinpointed the problem, will be fixed soon.

**Comment by kliuless** [comment]
It's simple to reproduce:

    class Bar
      include Contracts
    
      Contract Num, Num => Num
      def add(a,b)
        a + b
      end
    end
    
    p Bar.new.add(1,2)      # ok
    p Bar.new.add('a', 'b') # error

**Comment by waterlink** [comment]
Yes, it is discouraged and it is the root of the problem, we have it included at the top level in specs and were unaware of these problems..

**Comment by waterlink** [comment]
Attached PR should fix that, but I still fighting with 1.8.7 to make it work

**Comment by alex-fedorov** [comment]
@kliuless fix is in master, you can try it out.
